File: Samples/Basic/DocumentReader-sample/app/src/main/java/com/regula/documentreader/LicenseUtil.java
package com.regula.documentreader;

import android.app.Activity;

import androidx.annotation.NonNull;

import java.io.InputStream;

/**
 * Created by Sergey Yakimchik on 10.09.21.
 * Copyright (c) 2021 Regula. All rights reserved.
 */

class LicenseUtil {

    /**
     * Reading the license from raw resource file
     * @param activity current activity
     * @return license byte array
     */
    public static byte[] getLicense(@NonNull Activity activity) {
        try {
            InputStream licInput = activity.getResources().openRawResource(R.raw.regula);
            int available = licInput.available();
            final byte[] license = new byte[available];
            //noinspection ResultOfMethodCallIgnored
            licInput.read(license);

            licInput.close();

            return license;
        } catch (Exception ex) {
            ex.printStackTrace();
        }

        return null;
    }
}

File: Samples/Basic/DocumentReader-sample_kotlin/app/src/main/java/com/regula/documentreader/LicenseUtil.kt
package com.regula.documentreader

import android.app.Activity

/**
 * Created by Sergey Yakimchik on 10.09.21.
 * Copyright (c) 2021 Regula. All rights reserved.
 */
internal object LicenseUtil {
    /**
     * Reading the license from raw resource file
     * @param activity current activity
     * @return license byte array
     */
    fun getLicense(activity: Activity): ByteArray? {
        try {
            val licInput = activity.resources.openRawResource(R.raw.regula)
            val available = licInput.available()
            val license = ByteArray(available)
            licInput.read(license)
            licInput.close()
            return license
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        return null
    }
}